src/particles/Push.cpp

```diff
@@ -138,14 +138,16 @@ namespace detail
                 // here we just access the element by its respective type
                 std::visit(
                     [=, &ref_part](auto element) {
+
+                        // push reference particle in global coordinates
+                        element(ref_part);
+
```

Comment:
Note:  For each step, it is now essential to push the reference particle _before_ pushing the beam.  This is because "element(ref_part)" integrates the reference trajectory and the linear map equations simultaneously, the latter which are used to push the beam (for the case of the RF cavity).  This change of order should not affect any elements other than the RF cavity.